apps/web/src/server/actions/organization.ts

```diff
@@ -34,36 +76,311 @@ export async function getOrganizationProjects(id: Organization["id"]) {
+  } catch (e) {
+    throw new Error("Organization name already exists");
   }
+}
+
+export async function updateOrganizationName({
+  oldName,
+  name,
+}: {
+  oldName: string;
+  name: string;
+}): Promise<null | Error> {
+  const session = await getServerAuthSession();
+
+  const owner = (
+    await db
+      .select()
+      .from(organizationWithMembersView)
+      .where(
+        and(
+          eq(organizationWithMembersView.name, oldName),
+          eq(organizationWithMembersView.memberRole, "owner"),
+        ),
+      )
+  )[0]!;
+
+  if (owner.memberId !== session!.user.id) {
+    throw new Error(
+      "You must be the owner of the organization to perform this action",
+    );
+  }
+
+  await db
+    .update(organizations)
+    .set({ name })
+    .where(and(eq(organizations.name, oldName)));
 
-  return db
-    .select({
-      orgId: organizations.id,
-      orgName: organizations.name,
-      orgOwner: organizations.owner,
-    })
+  return null;
+}
+
+export async function deleteOrganization(
+  name: Organization["name"],
+): Promise<null | Error> {
+  const session = await getServerAuthSession();
+
+  const orgWithMember = (
+    await db
+      .select()
+      .from(organizationWithMembersView)
+      .where(
+        and(
+          eq(organizationWithMembersView.memberRole, "owner"),
+          eq(organizationWithMembersView.name, name),
+        ),
+      )
+  )[0]!;
+
+  if (orgWithMember.memberId !== session!.user.id) {
+    throw new Error(
+      "You must be the owner of the organization to perform this action",
+    );
+  }
+
+  const currentUserOrganizations = await db
+    .select()
     .from(usersToOrganizations)
+    .where(eq(usersToOrganizations.memberId, session!.user.id))
+    .execute();
+
+  if (currentUserOrganizations.length < 2) {
+    throw new Error("You must be a member of at least one organization.");
+  }
+
+  await db
+    .delete(usersToOrganizations)
+    .where(eq(usersToOrganizations.organizationId, orgWithMember.id));
+
+  await db.delete(organizations).where(eq(organizations.name, name));
+
+  return null;
+}
+
+export async function getMembersOfOrganization(id: Organization["id"]) {
+  const members = await db
+    .select()
+    .from(usersToOrganizations)
+    .leftJoin(users, eq(users.id, usersToOrganizations.memberId))
     .leftJoin(
       organizations,
-      eq(usersToOrganizations.organizationId, organizations.id),
+      eq(organizations.id, usersToOrganizations.organizationId),
     )
-    .where(eq(usersToOrganizations.memberId, session.user.id));
+    .where(or(eq(usersToOrganizations.organizationId, id)))
+    .execute();
+
+  return members;
 }
 
-export async function createOrganization({ name }: InsertOrganization) {
-  const session = await getServerAuthSession();
+export async function addMemberToOrganization({
+  organizationId,
+  memberId,
+  role,
+}: {
+  organizationId: Organization["id"];
+  memberId: string;
+  role: "user" | "owner" | "admin";
```

Comment:
Change this to a Role union type from `validators/organization.ts`

Reply:
done